src/import-data/ImportData.test.ts

```diff
@@ -0,0 +1,448 @@
+
+  asMockedFn(useRoute).mockReturnValue({
+    query: queryParams,
+  });
+
+  render(h(ImportData));
+};
+
+const importIntoExistingWorkspace = async (user: UserEvent, workspaceName: string): Promise<void> => {
+  const existingWorkspace = screen.getByText('Start with an existing workspace', { exact: false });
+  await user.click(existingWorkspace);
+
+  const workspaceSelect = new SelectHelper(screen.getByLabelText('Select a workspace'), user);
+  await workspaceSelect.selectOption(workspaceName);
+
+  await user.click(screen.getByRole('button', { name: 'Import' }));
+};
+
+describe('ImportData', () => {
+  beforeEach(() => {
+    // Arrange
+    asMockedFn(useWorkspaces).mockReturnValue({
+      workspaces: [defaultAzureWorkspace, defaultGoogleWorkspace],
+      loading: false,
+      refresh: () => Promise.resolve(),
+    });
+
+    asMockedFn(useDataCatalog).mockReturnValue({
+      dataCatalog: [],
+      loading: false,
+      refresh: () => Promise.resolve(),
+    });
+  });
+
+  describe('PFB imports', () => {
+    it('imports PFB files', async () => {
+      // Arrange
+      const user = userEvent.setup();
+
+      const importJob = jest.fn().mockResolvedValue({ jobId: 'new-job' });
+      const getWorkspaceApi = jest.fn().mockReturnValue({
+        importJob,
+      });
+
+      const importUrl = 'https://example.com/path/to/file.pfb';
+      setup({
+        mockAjax: {
+          Workspaces: {
+            workspace: getWorkspaceApi,
+          },
+        },
+        queryParams: {
+          format: 'PFB',
+          url: importUrl,
+        },
+      });
+
+      // Act
+      await importIntoExistingWorkspace(user, defaultGoogleWorkspace.workspace.name);
+
+      // Assert
+      expect(getWorkspaceApi).toHaveBeenCalledWith(
+        defaultGoogleWorkspace.workspace.namespace,
+        defaultGoogleWorkspace.workspace.name
+      );
+
+      expect(importJob).toHaveBeenCalledWith(importUrl, 'pfb', null);
+    });
+  });
+
+  describe('BagIt imports', () => {
+    it('imports BagIt file when format is unspecified', async () => {
+      // Arrange
+      const user = userEvent.setup();
+
+      const importBagit = jest.fn().mockResolvedValue(undefined);
+      const getWorkspaceApi = jest.fn().mockReturnValue({
+        importBagit,
+      });
+
+      const importUrl = 'https://example.com/path/to/file.bagit';
+      setup({
+        mockAjax: {
+          Workspaces: {
+            workspace: getWorkspaceApi,
+          },
+        },
+        queryParams: {
+          url: importUrl,
+        },
+      });
+
+      // Act
+      await importIntoExistingWorkspace(user, defaultGoogleWorkspace.workspace.name);
+
+      // Assert
+      expect(getWorkspaceApi).toHaveBeenCalledWith(
+        defaultGoogleWorkspace.workspace.namespace,
+        defaultGoogleWorkspace.workspace.name
+      );
+
+      expect(importBagit).toHaveBeenCalledWith(importUrl);
+    });
+  });
+
+  describe('Entities JSON imports', () => {
```

Comment:
drop `describe` and outdent? (and only re-add if you end up with more cases like TDR imports)